tests/sota_tools/treehub_server.py

```diff
@@ -79,6 +79,17 @@ def do_POST(self):
             self.send_response_only(200)
             self.end_headers()
             return
+        elif ctype == 'application/octet-stream':
```

Comment:
In case of 'application/octet-stream' we can read the posted file/body just from the socket buffer, so something like that should work

       ```
 elif ctype == 'application/octet-stream':
            length = int(self.headers['content-length'])
            body = self.rfile.read(length)
            with open("body.dat", "wb") as f:
                f.write(body)

            self.send_response_only(200)
            self.end_headers()
```

But, if the given implementation works then it's ok - this is just a test tool

Reply:
Thanks Mike, I'll try your way. I was also just gonna try this:

import cgitb; cgitb.enable()

form = cgi.FieldStorage()
fileitem = form['filename']
if fileitem.filename:
   with open("body.dat", "wb") as f:
   f.write(fileitem.file.read())

Reply:
Tested Mike's way, t_deploy output:

diff: /tmp/aktualizr-951f-8561-2d1e-5c0a/d5ea-2beb-dir/objects/: No such file or directory
/home/liping/aktualizr/src/sota_tools/deploy_test.cc:32: Failure

Reply:
Interesting, I've just checked and it works for me, I missed an ending `return` statement at the end of the code snippet

Reply:
Yes, I added the return statement. maybe we can see how it work out In CI test?

Reply:
Oh, I think, I understand where is a problem. It's not enough just set a header that indicates a type of transferring data, actual data should be transferred according to the declared way. In this particular case we set 'application/octet-stream' but actually transfer data as `'multipart/form-data'`  see OSTreeObject::Upload() code below setting the headers.